tests/e2e/support/objects/app-files/resource/actions.ts

```diff
@@ -240,7 +252,94 @@ export const createNewFileOrFolder = async (args: createResourceArgs): Promise<v
       await page.goto(page.url())
       break
     }
+    case 'OpenDocument': {
+      // By Default when OpenDocument is created, it is opened with collabora if both app-provider services are running together
+      await createOpenDocumentOrMicrosoftWordFile(args, 'Collabora')
+      break
+    }
+    case 'Microsoft Word': {
+      // By Default when Microsoft Word document is created, it is opened with OnlyOffice if both app-provider services are running together
+      await createOpenDocumentOrMicrosoftWordFile(args, 'OnlyOffice')
+      break
+    }
+  }
+}
+
+const createOpenDocumentOrMicrosoftWordFile = async (
+  args: createResourceArgs,
+  editorToOpen: string
+): Promise<void> => {
+  const { page, name, type, content } = args
+  await page.locator(util.format(createNewOpenDocumentFileButton, type)).click()
+  await page.locator(resourceNameInput).fill(name)
+  let newEditorTab
+  await Promise.all([
+    (newEditorTab = page.waitForEvent('popup')),
+    page.waitForResponse((resp) => resp.status() === 200 && resp.request().method() === 'POST'),
+    page.locator(util.format(actionConfirmationButton, 'Create')).click()
+  ])
+  const editorPage = await newEditorTab
```

Comment:
can be deconstructed
```suggestion
  const [ editorPage ] = await Promise.all([
    page.waitForEvent('popup'),
    page.waitForResponse((resp) => resp.status() === 200 && resp.request().method() === 'POST'),
    page.locator(util.format(actionConfirmationButton, 'Create')).click()
  ])
```